sparse_strips/vello_macros/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+}
+
+struct Arguments {
+    /// The width of the scene.
+    width: u16,
+    /// The height of the scene.
+    height: u16,
+    /// The (additional) maximum tolerance for how much two pixels are allowed to deviate from each other
+    /// when comparing to the reference images. Some renderers already have an existing tolerance
+    /// (currently 0 for the CPU and 1 for the hybrid renderer), this value will simply be added
+    /// to the currently existing threshold
+    threshold: u8,
+    /// Whether the background should be transparent.
+    transparent: bool,
+    /// Whether the test should not be run on the CPU (`vello_cpu`).
+    skip_cpu: bool,
+    /// Whether the test should not be run on the GPU (`vello_hybrid`).
+    skip_gpu: bool,
+    /// Whether no reference image should actually be created (for tests that only check
+    /// for panics, but are not interested in the actual output).
+    no_ref: bool,
+}
+
+impl Default for Arguments {
+    fn default() -> Self {
+        Self {
+            width: 100,
+            height: 100,
+            threshold: 0,
+            transparent: false,
+            skip_cpu: false,
+            skip_gpu: false,
+            no_ref: false,
+        }
+    }
+}
+
+/// Create a new vello snapshot test.
+#[proc_macro_attribute]
+pub fn v_test(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let attrs = parse_macro_input!(attr as AttributeInput);
+
+    let input_fn = parse_macro_input!(item as ItemFn);
+
+    let input_fn_name = input_fn.sig.ident.clone();
+    let u8_fn_name = Ident::new(&format!("{}_cpu_u8", input_fn_name), input_fn_name.span());
+    let hybrid_fn_name = Ident::new(&format!("{}_gpu", input_fn_name), input_fn_name.span());
+
+    let input_fn_name_str = input_fn_name.to_string();
+    let u8_fn_name_str = u8_fn_name.to_string();
+    let hybrid_fn_name_str = hybrid_fn_name.to_string();
+
+    let Arguments {
+        width,
+        height,
+        threshold,
+        transparent,
+        skip_cpu,
+        mut skip_gpu,
+        no_ref,
+    } = parse_args(&attrs);
+
+    // These tests currently don't work with `vello_hybrid`.
+    skip_gpu |= {
+        input_fn_name_str.contains("clip")
+            || input_fn_name_str.contains("compose")
+            || input_fn_name_str.contains("gradient")
+            || input_fn_name_str.contains("image")
+            || input_fn_name_str.contains("layer")
+            || input_fn_name_str.contains("mask")
+            || input_fn_name_str.contains("mix")
+            || input_fn_name_str.contains("opacity")
+    };
+
+    let empty_snippet = quote! {};
+    let ignore_snippet = quote! {#[ignore]};
+
+    let ignore_hybrid = if skip_gpu {
+        ignore_snippet.clone()
+    } else {
+        empty_snippet.clone()
+    };
+    let ignore_cpu = if skip_cpu {
+        ignore_snippet.clone()
+    } else {
+        empty_snippet.clone()
+    };
+    let cpu_threshold = threshold;
+    let gpu_threshold = threshold + 1;
```

Comment:
Might be worth defining these magic values to the top of the file as constants? 

```
const DEFAULT_CPU_TOLERANCE: u32 = 0;
const DEFAULT_HYBRID_TOLERANCE: u32 = 1;
```

and then refer to them in the documentation above threshold